File: HW4/hw4.py
# this is Homework for CSE 465
# Author Bohan Feng

import sys
import csv

def fileOutput(filename, str):
    '''
        this method create a file output
        Args:
            filename (str): out filename
            str (str): content of outputfilne
    '''
    extname = ".txt"
    filename += extname
    f = open(filename, "w")
    f.write(str)
# end def

def modifieLine(line, list):
    '''
        this method replace the '<<key>>' to "<<value>>"
        Args: 
            line (str): a string to be modified
            list (dict): a dictionary that contains the keys and value
        Return:
            line (str): a modified string
    ''' 
    for x in list:
        line = line.replace("<<" + x + ">>", list[x])
    # end for
    return line
# end def

def getresult(filename1, filename2):
    '''
        this method reads files and get the content and make the content into specic format
        Args:
            filename1 (str): tsv file
            finename2 (str): tmp file
    '''
    with open(filename1) as tsvfile:
        reader = csv.DictReader(tsvfile, dialect='excel-tab')
        for row in reader:
            re = ""
            with open(filename2) as f:
                for line in f:
                    re += modifieLine(line, row)
                # end for
            # end with
            fileOutput(row['ID'], re)
        # end for
    # end with
# end def

# filename
filename1 = sys.argv[1]
filename2 = sys.argv[2]

# get the result
getresult(filename1, filename2)

File: HW4/hw4.cs
/*
    this is Homework for CSE 465
    Author Bohan Feng
 */
using System;
using System.IO;

public class Homework04{

    /**
        this method create a file output
        Args:
            filename (str): out filename
            str (str): content of outputfilne
     */
    public static void fileOutput(String filename, String str){
        using(StreamWriter sw = new StreamWriter(filename+".txt")){
            sw.WriteLine(str);
        }
    }

    /**
        this method replace the '<<key>>' to "<<value>>"
        Args: 
            line (str): a string to be modified
            keys (String[]): a array the contain the header of tsv file
            values (Stirng[]) a array contain rows of tsv file
        Return:
            line (str): a modified string
     */
    public static string modifieLine(String line, String[] keys, String[] values) {
        for(int i = 0; i < keys.Length; ++i){
            line = line.Replace("<<" + keys[i] + ">>", values[i]);
        }
        return line;
    }

    /**
        this method reads files and get the content and make the content into specic format
        Args:
            filename1 (str): tsv file
            finename2 (str): tmp file
     */
    public static void getResult(String filename1, String filename2){
        //use streamReader from IO to create object 
        StreamReader tsvFile = new StreamReader(filename1);
        StreamReader tmpFile = new StreamReader(filename2);
		//list keys for content split by tab
        String[] keys = tsvFile.ReadLine().Split('\t');
        String temp = tmpFile.ReadToEnd();

        String line = "";
        while((line = tsvFile.ReadLine()) != null){
            String[] values = line.Split('\t');
            int idIndex = Array.IndexOf(keys,"ID");
            String re = temp;
            re = modifieLine(re, keys, values);
            fileOutput(values[idIndex], re);
        }
    }

    public static void Main(String[] args)
    {
        // file name
        String filename1 = args[0];
        String filename2 = args[1];
        // get the result
        getResult(filename1, filename2);
    } 
}